src/kernel.hpp

```diff
@@ -237,7 +245,8 @@ struct cvk_kernel_argument_values {
         }
 
         if (m_entry_point->has_pod_arguments() ||
-            m_entry_point->has_image_metadata()) {
+            m_entry_point->has_image_metadata() ||
+            m_entry_point->has_sampler_metadata()) {
```

Comment:
It's really time we refactor push constant management to make it more readable. In a lot of places, they're still named "POD". Best done as a standalone changes.